File: examples/js/dbusBridge.js
/*
 * dbusBridge.js
 *
 * EventDance examples
 *
 * Copyright (C) 2011-2012, Igalia S.L.
 *
 * Authors:
 *   Eduardo Lima Mitev <elima@igalia.com>
 */

imports.searchPath.unshift ("../common");

const Gio = imports.gi.Gio;
const Evd = imports.gi.Evd;

const LISTEN_PORT = 8080;
const DBUS_ADDR = "alias:abstract=/org/eventdance/lib/examples/dbus-bridge";

// Evd daemon
var daemon = new Evd.Daemon.get_default (null, null);

// Session bus addr
var sessionBusAddr = Gio.dbus_address_get_for_bus_sync(Gio.BusType.SESSION, null);

// Web transport
var transport = new Evd.WebTransportServer ();

function onNewPeer (transport, peer) {
    // This is to send a virtual DBus address to the peer instead of the real DBus daemon
    // address, for consistency and security reasons.
    Evd.dbus_agent_create_address_alias (peer, sessionBusAddr, DBUS_ADDR);
}

if (transport["signal"])
    transport.signal.new_peer.connect (onNewPeer);
else
    transport.connect ("new-peer", onNewPeer);

// DBus bridge
var dbusBridge = new Evd.DBusBridge ();
dbusBridge.add_transport (transport);

// Web dir
var webDir = new Evd.WebDir ({ root: "../common" });

// Web selector
var selector = new Evd.WebSelector ();
selector.set_default_service (webDir);
transport.set_selector (selector);

// start listening
selector.listen ("0.0.0.0:" + LISTEN_PORT, null,
    function (service, result) {
        try {
            service.listen_finish (result);
            print ("Listening on port " + LISTEN_PORT + ", now point your browser to any of the DBus example web pages");
        }
        catch (e) {
            print ("Error: " + e);
            daemon.quit (-1);
        }
    }, null);

// start the show
daemon.run ();

File: examples/python/dbus-bridge.py
#
# dbusBridge.js
#
# EventDance examples
#
# Copyright (C) 2011, Igalia S.L.
#
# Authors:
#   Eduardo Lima Mitev <elima@igalia.com>
#

import gobject
from gi.repository import Gio
from gi.repository import Evd

LISTEN_PORT = '8080'
DBUS_ADDR = 'alias:abstract=/org/eventdance/lib/examples/dbus-bridge'

# Session bus address
session_bus_addr = Gio.dbus_address_get_for_bus_sync(Gio.BusType.SESSION, None)

# Web transport
transport = Evd.WebTransport()

def on_new_peer(transport, peer, main_loop):
    # This is to send a virtual DBus address to the peer instead of the real DBus daemon
    # address, for consistency and security reasons.
    Evd.dbus_agent_create_address_alias (peer, session_bus_addr, DBUS_ADDR);

transport.connect('new-peer', on_new_peer, None)

# DBus bridge
dbus_bridge = Evd.DBusBridge()
dbus_bridge.add_transport(transport)

# Web dir
web_dir = Evd.WebDir()
web_dir.set_root ('../common')

# Main loop
main_loop = gobject.MainLoop();

# Web selector
selector = Evd.WebSelector()
selector.set_default_service(web_dir)
transport.set_selector(selector);

def on_listen(self, result, main_loop):
    try:
        self.listen_finish(result)
        print ('Listening on port ' + LISTEN_PORT + ', now point your browser to any of the DBus example web pages')
    except Exception as e:
        print('Error: ' + str(e))
        main_loop.quit()

# start listening
selector.listen_async('0.0.0.0:' + LISTEN_PORT, None, on_listen, main_loop)

# start the show
main_loop.run();
